src/common/transformations/src/transformations/symbolic_transformations/symbol_optimization.cpp

```diff
@@ -354,12 +355,85 @@ void save_shape_sources(const std::shared_ptr<ov::Node>& op, STS_map& symbol_sha
         }
     }
 }
+
+struct OutputValue {
+    std::vector<ov::Any> value;
+
+    bool operator==(const OutputValue& other) const {
+        return value == other.value;
+    }
+
+    bool operator<(const OutputValue& other) const {
+        return std::lexicographical_compare(
+            std::begin(value),
+            std::end(value),
+            std::begin(other.value),
+            std::end(other.value),
+            [](const ov::Any& a, const ov::Any& b) {
+                // each element is either a symbol or an integer. in case they differ any integer is less than a symbol.
+                if (a.is<std::shared_ptr<ov::Symbol>>() && b.is<std::shared_ptr<ov::Symbol>>())
+                    return a.as<std::shared_ptr<ov::Symbol>>() < b.as<std::shared_ptr<ov::Symbol>>();
+                if (a.is<int64_t>() && b.is<int64_t>())
+                    return a.as<int64_t>() < b.as<int64_t>();
+                return a.is<int64_t>();
+            });
+    }
+
+    static std::pair<bool, OutputValue> make(const ov::Output<ov::Node>& output) {
```

Comment:
Maybe use `ov::optional` from `ov_optional.hpp` or `unique_ptr` ?
```suggestion
    static ov::optional<OutputValue> make(const ov::Output<ov::Node>& output) {
```